### PR TITLE
[REF] base_geolocalize: support multiple provider

### PR DESCRIPTION
**Pull request by JKE-be**
- Make it generic: geo_coder (logic) / geo_provider (list of provider)
- Add support for openstreetmap since google make it more complicated: https://cloud.google.com/maps-platform/user-guide/account-changes

Courtesy of @ecino

### CODE REVIEW

File: addons/base_geolocalize/__manifest__.py [hunk 1]
```diff
@@ -2,15 +2,18 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 {
     'name': 'Partners Geolocation',
-    'version': '2.0',
-    'category': 'Sales',
+    'version': '2.1',
+    'category': 'Extra Tools',
     'description': """
```
Comment (pedrobaeza) [inline addons/base_geolocalize/__manifest__.py]: You should remove this description for applying the README.md one

**Comment by JKE-be** [comment]
Robodoo r+

**Comment by JKE-be** [comment]
Robodoo retry

**Comment by JKE-be** [comment]
Robodoo all what you want

**Comment by rdeodoo** [comment]
@JKE-be Maybe @robodoo doesn't like to be called 'robodoo retry' with an uppercase :grin: ([the syntax here is the same without uppercase](https://github.com/odoo/odoo/pull/27735#issuecomment-432374455))?

**Comment by KangOl** [comment]
@robodoo listen to jke and retry

**Comment by JKE-be** [comment]
Oufffff thx all for your help

**Pull request closed without merging**